flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -468,6 +470,8 @@ private void validateClusterSpecification(ClusterSpecification clusterSpecificat
 			flinkConfiguration.setString(dynProperty.getKey(), dynProperty.getValue());
 		}
 
+		this.flinkVersion = dynProperties.getOrDefault("flink-version", "");
```

Comment:
Whether we could get the version from the properties already in this class, like `flinkJarPath`?

Reply:
For most of the cases, we can get the precise version of flinkJarPath. But for some deployment systems that submit jobs from customers, different jar will be used. Probably, using dynamic properties can give more flexibility here. How do you think?

Reply:
Do you want to show the same version for different jars?

Reply:
No, different versions can be specified by the dynamic properties right? It can be either through cmd or the backend service that launches flink job.